### PR TITLE
DOCS-2031: Add arm, base, board example snippets

### PR DESCRIPTION
**Pull request by JessamyT**
Not sure of syntax on many of these Viam-specific types--apologies if this is super wrong.

### CODE REVIEW

File: lib/src/components/board/board.dart [hunk 1]
```diff
@@ -15,39 +15,101 @@ abstract class Board extends Resource {
...
   /// Get the PWM frequency of the given pin of a board.
+  ///
+  /// ```
+  /// // Get the PWM frequency of pin 11
+  /// var frequency = await myBoard.get("11");
+  /// ```
   Future<int> pwmFrequency(String pin, {Map<String, dynamic>? extra});
 
-  /// Set the PWM frequency of the given pin of a board.
+  /// Set the PWM frequency in Hertz of the given pin of a board.
```
Comment (JessamyT) [inline lib/src/components/board/board.dart:57]:
```suggestion
  /// Set the PWM frequency in hertz of the given pin of a board.
```

File: lib/src/components/base/base.dart [hunk 2]
```diff
@@ -29,15 +39,45 @@ abstract class Base extends Resource {
   /// When both [linear] and [angular] are zero-vectors, the base will stop.
   /// When [linear] and [angular] are both non-zero-vectors, the base will move in an arc,
   /// with a tighter radius if [angular] power is greater than [linear] power.
```
Comment (JessamyT) [inline lib/src/components/base/base.dart:41]:
```suggestion
  /// with a tighter radius if [angular] power is greater than [linear] power.
  /// The positive y axis points forwards and the positive z axis points up.
```

File: lib/src/components/board/board.dart [hunk 3]
```diff
@@ -15,39 +15,101 @@ abstract class Board extends Resource {
...
+  /// // Get the PWM duty cycle of pin 15
+  /// var dutyCycle = await myBoard.pwm('15');
+  /// ```
   Future<double> pwm(String pin, {Map<String, dynamic>? extra});
 
   /// Set the duty cycle of the given pin of a board.
+  ///
+  /// ```
+  /// // Set the PWM duty cycle of pin 13
+  /// await myBoard.setPwm('13');
```
[the remark this answers is not in the document]
Comment (JessamyT) [reply lib/src/components/board/board.dart:45]:
```suggestion
  /// await myBoard.setPwm('13', 0.6);
```

**Review by njooma — COMMENTED** [review]
Can you provide a screenshot of how this looks when it's rendered? You can do so by running `dart doc` in your terminal from the flutter-sdk project directory, and then opening the generated page `doc/api/index.html` in your browser and navigating to one of these pages

**Comment by JessamyT** [comment]
image

**Review by njooma — APPROVED** [review]

File: lib/src/components/base/base.dart (hunk 2 above)
Comment (stuqdog) [reply lib/src/components/base/base.dart:41]: I think so, yes. Thanks!

**Review by stuqdog — APPROVED** [review]

File: lib/src/components/arm/arm.dart [hunk 4]
```diff
@@ -7,24 +7,57 @@ abstract class Arm extends Resource {
   static const Subtype subtype = Subtype(resourceNamespaceRDK, resourceTypeComponent, 'arm');
 
   /// Get the current [Pose] of the end of the arm.
+  ///
+  /// ```
+  /// // Get the pose of an arm named "myArm"
+  /// const currentPose = await myArm.endPosition();
```
Comment (JessamyT) [inline lib/src/components/arm/arm.dart:13]:
```suggestion
  /// final currentPose = await myArm.endPosition();
```

File: lib/src/components/arm/arm.dart [hunk 5]
```diff
@@ -7,24 +7,57 @@ abstract class Arm extends Resource {
...
+  /// ```
   Future<Pose> endPosition({Map<String, dynamic>? extra});
 
   /// Move the end of the arm to the [Pose] specified.
+  ///
+  /// ```
+  /// // Create a pose for the arm to move to
+  /// final Pose targetPose = Pose(12, 0, 400, 0, 0, 1);
```
Comment (JessamyT) [inline lib/src/components/arm/arm.dart:21]:
```suggestion
  /// final targetPose = Pose.fromBuffer([12, 0, 400, 0, 0, 1, 90]);
```

File: lib/src/components/arm/arm.dart [hunk 6]
```diff
@@ -7,24 +7,57 @@ abstract class Arm extends Resource {
...
 
-  /// Move each joint on the arm to the corresponding position specified in [positions]
+  /// Move each joint on the arm to the corresponding position specified in [positions].
+  ///
+  /// ```
+  /// // Create a list of joint angles for each arm joint
+  /// List<double> targetPositions = [180, 90, 15.75, 30, 90, 0];
+  ///
+  /// // Move the arm joints to those angles
+  /// await myArm.moveToJointPosition(targetPositions);
```
Comment (JessamyT) [inline lib/src/components/arm/arm.dart:35]:
```suggestion
  /// await myArm.moveToJointPositions(targetPositions);
```

**Pull request merged**